Modules/_ctypes/_ctypes.c

```diff
@@ -1948,33 +2064,24 @@ PyCSimpleType_paramfunc(CDataObject *self)
     return parg;
 }
 
-static PyObject *
-PyCSimpleType_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
+static int
+PyCSimpleType_init(PyObject *self, PyObject *args, PyObject *kwds)
 {
-    PyTypeObject *result;
-    StgDictObject *stgdict;
     PyObject *proto;
     const char *proto_str;
     Py_ssize_t proto_len;
     PyMethodDef *ml;
     struct fielddesc *fmt;
 
```

Comment:
```suggestion

    if (PyType_Type.tp_init(self, args, kwds) < 0) {
        return -1;
    }
```
This can emit `TypeError: type.__init__() takes 1 or 3 arguments`, which would help `CreateSwappedType()` avoid a crash.  An exaggerated version is a779f16e8e9b18838823754b33280993b0cc7c95.

Reply:
Thanks! I also added the test from your PR.